scipy/optimize/optimize.py

```diff
@@ -1720,9 +1720,8 @@ def fminbound(func, x1, x2, args=(), xtol=1e-5, maxfun=500,
         return res['x']
 
 
-def _minimize_scalar_bounded(func, bounds, args=(),
-                             xatol=1e-5, maxiter=500, disp=0,
-                             **unknown_options):
+def _minimize_scalar_bounded(func, bounds, args=(), xatol=1e-5, maxiter=500,
+                             disp=False, **unknown_options):
```

Comment:
Unfortunately, this would break how verbosity is controlled in `fminbound()` which also uses this function (though, perhaps there's no test coverage for this).
So `disp` should remain an integer here (and changes below be reverted). Instead, `minimize()` should be fixed to call `_minimize_scalar_bounded()` with `disp` as an integer obtained from the `bool` value it receives. For instance:
```
diff --git a/scipy/optimize/_minimize.py b/scipy/optimize/_minimize.py
--- a/scipy/optimize/_minimize.py
+++ b/scipy/optimize/_minimize.py
@@ -655,6 +655,11 @@ def minimize_scalar(fun, bracket=None, bounds=None, args=(),
         if bounds is None:
             raise ValueError('The `bounds` parameter is mandatory for '
                              'method `bounded`.')
+        # replace boolean "disp" option, if specified, by an integer value, as
+        # expected by _minimize_scalar_bounded()
+        disp = options.get('disp')
+        if isinstance(disp, bool):
+            options['disp'] = 2 * int(disp)
         return _minimize_scalar_bounded(fun, bounds, args, **options)
     elif meth == 'golden':
         return _minimize_scalar_golden(fun, bracket, args, **options)
```

Reply:
Thanks for pointing this out, I reverted the commit and changed the code as suggested. I also corrected the docstring in _minimize_scalar_bounded (now stating that disp is int instead of bool).